### PR TITLE
Using pushlog to validate revisions

### PR DESCRIPTION
**Pull request by adusca**
I still need to update the tests, but this is the general idea. Since we discovered that a revision doesn't have to exist on buildapi for trigger_arbitrary_job to work (see comments in #313 ).

This allows us to trigger stuff in revisions on try with only TC jobs and in DONTBUILD revisions.

### CODE REVIEW

File: mozci/sources/buildapi.py [hunk 1]
```diff
@@ -157,49 +157,23 @@ def _valid_builder():
                     "mozci.allthethings.")
 
 
-def valid_revision(repo_name, revision):
```
Comment (armenzg) [inline mozci/sources/buildapi.py:160]: We might still want to keep this around (but with a different name) to validate credentials of a user as soon as possible.

File: mozci/scripts/triggerbyfilters.py [hunk 2]
```diff
@@ -75,7 +76,8 @@ def main():
         LOG = setup_logging(logging.INFO)
 
     if options.rev == 'tip':
-        options.rev = query_repo_tip(options.repo)
+        repo_url = query_repo_url(options.repo)
+        options.rev = query_repo_tip(repo_url)
```
Comment (armenzg) [inline mozci/scripts/triggerbyfilters.py:81]: The day we decide to move pushlog as its own client we might want to teach it how to determine the repo_url by itself.
Thank you for removing the dependency on buildapi.

File: mozci/sources/buildapi.py (hunk 1 above)
Comment (armenzg) [inline mozci/sources/buildapi.py:160]: More clearly, rename the function to validate credentials and use another end-point which will always be available (unless this one with a revision).

**Comment by armenzg** [comment]
All looking good.
I need that removed function to be addressed.
Merge once satisfied. Good job!

File: mozci/sources/buildapi.py (hunk 1 above)
Comment (adusca) [inline mozci/sources/buildapi.py:160]: I added a function called valid_crendentials that calls get_credentials and raises an AuthenticationError when the credentials are invalid, but I'm not sure where we should call this function.

Comment (adusca) [inline mozci/sources/buildapi.py:160]: I'm calling valid_credentials as the first step in both scripts, so we ask for credentials as soon as possible.

**Comment by armenzg** [comment]
Beautiful :+1:

**Pull request merged**